akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/OnSignalSpec.scala

```diff
@@ -29,5 +29,36 @@ final class OnSignalSpec extends ScalaTestWithActorTestKit with AnyWordSpecLike
       spawn[Nothing](behavior)
       probe.expectMessage(Done)
     }
+
+    def stopper(probe: TestProbe[Done]) =
+      Behaviors
+        .receiveMessage[String] {
+          case "stop" =>
+            Behaviors.stopped
+        }
+        .receiveSignal {
+          case (_, PostStop) =>
+            probe.ref ! Done
+            Behaviors.same
+        }
```

Comment:
Some additional test coverage by adding alternative test where this also has a child. That is handled differently in the implementation because then it waits for the children to be terminated before emitting the PostStop signal.